packages/integrations/src/tracing.ts

```diff
@@ -0,0 +1,170 @@
+import { EventProcessor, Hub, Integration } from '@sentry/types';
+import { fill, getGlobalObject, isMatchingPattern, SentryError, supportsNativeFetch } from '@sentry/utils';
+
+/** JSDoc */
+interface TracingOptions {
+  tracingOrigins: Array<string | RegExp>;
+  traceXHR?: boolean;
+  traceFetch?: boolean;
+  autoStartOnDomReady?: boolean;
+}
+
+/**
+ * Tracing Integration
+ */
+export class Tracing implements Integration {
+  /**
+   * @inheritDoc
+   */
+  public name: string = Tracing.id;
+
+  /**
+   * @inheritDoc
+   */
+  public static id: string = 'Tracing';
+
+  /**
+   * If we have an xhr we need to store the url in the instance.
+   *
+   */
+  // @ts-ignore
+  private _xhrUrl?: string;
+
+  /**
+   * Constructor for Tracing
+   *
+   * @param _options TracingOptions
+   */
+  public constructor(private readonly _options: TracingOptions) {
+    if (!_options.tracingOrigins || !Array.isArray(_options.tracingOrigins) || _options.tracingOrigins.length === 0) {
+      throw new SentryError('You need to define `tracingOrigins` in the options. Set an array of urls to trace.');
+    }
+  }
+
+  /**
+   * @inheritDoc
+   */
+  public setupOnce(_: (callback: EventProcessor) => void, getCurrentHub: () => Hub): void {
+    if (this._options.traceXHR !== false) {
+      this._traceXHR(getCurrentHub);
+    }
+    if (this._options.traceFetch !== false) {
+      this._traceFetch(getCurrentHub);
+    }
+    if (this._options.autoStartOnDomReady !== false) {
+      getGlobalObject<Window>().addEventListener('DOMContentLoaded', () => {
```

Comment:
There's a chance that DOM will already by parsed by the time this script runs (small, but there is, eg. when someone places script tag at the very bottom of the page).
We should check for the event AND using https://developer.mozilla.org/en-US/docs/Web/API/Document/readyState (whichever one is first wins)